output-comparator/src/main/java/org/mobilitydata/gtfsvalidator/outputcomparator/io/NoticeSummary.java

```diff
@@ -0,0 +1,82 @@
+/*
+ * Copyright 2020 MobilityData IO
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.mobilitydata.gtfsvalidator.outputcomparator.io;
+
+import com.google.gson.annotations.SerializedName;
+import java.util.Collections;
+import java.util.List;
+import java.util.Map;
+import org.mobilitydata.gtfsvalidator.notice.SeverityLevel;
+
+/**
+ * Used to deserialize a validation report. It is used to store information about one type of notice
+ * encountered in a validation report: error code, severity level, the total number of notices
+ * related to the error code and a list of notice contexts (which provides additional information
+ * about each notice.
+ */
+public class NoticeSummary {
+  private final String code;
+  private final SeverityLevel severity;
+
+  @SerializedName("totalNotices")
+  private final int count;
+
+  @SerializedName("notices")
+  private final List<Map<String, Object>> contexts;
```

Comment:
Can this be an ImmutableList of ImmutableMap? The getter and the constructor types will also change.

Reply:
Unfortunately, because we use a custom deserializer, this can not be immutable, however, the getter returns an immutable element to ensure `contexts` are immutable elements when retrieved from `getContext`.